dvc/_debug.py

```diff
@@ -70,6 +99,13 @@ def debugtools(args: "Namespace" = None, **kwargs):
             stack.enter_context(profile(kw.get("cprofile_dump")))
         if kw.get("instrument") or kw.get("instrument_open"):
             stack.enter_context(instrument(kw.get("instrument_open", False)))
+        if kw.get("yappi"):
+            from datetime import datetime
+
+            output = "callgrind.dvc-{0:%Y%m%d}_{0:%H%M%S}.out"
+            stack.enter_context(
+                yappi_profile(path=lambda: output.format(datetime.now()))
```

Comment:
```suggestion
                yappi_profile(path_factory=lambda: output.format(datetime.now()))
```
?

Reply:
yeah, we could do that too, but I am optimizing the API here. Note that this `yappi_profile` (along with all the others) can be used as a decorator and a context manager on any functions that you want to profile:

```python
c = count(start=1)


@yappi_profile(path=lambda: next(c))
def _diff(*args, **kwargs):
    pass
```